search-elasticsearch/es.go

```diff
@@ -246,8 +246,10 @@ func (s *SearchEngine) buildQuery(cond *plugin.SearchBasicCond) (
 	log.Debugf("build query: %+v", cond)
 
 	q := elastic.NewBoolQuery()
-	if len(cond.TagIDs) > 0 {
-		q.Must(elastic.NewTermsQuery("tags", convertToInterfaceSlice(cond.TagIDs)...))
+	for _, tagGroup := range cond.TagIDs {
+		if len(tagGroup) > 0 {
+			q.Must(elastic.NewTermsQuery("tags", convertToInterfaceSlice(tagGroup)...))
```

Comment:
@LinkinStars 
Here I'm not sure if calling the same logic multiple times can achieve the desired result.

Reply:
Hi, I've tested that it can achieve the result which is align with the logic in answer. For example, it'll generate below query.
```json
GET /answer_post/_search
{
  "query": {
    "bool": {
      "must": [
        {
          "terms": {
            "tags": [
              "10030000000000002", "10030000000000001"
            ]
          }
        },
        {
          "terms": {
            "tags": [
              "10030000000000031"
            ]
          }
        }
      ]
    }
  }
}
```

it means the doc contains tags with one of "10030000000000002", "10030000000000001" and contains "10030000000000031". You can find `terms` description here.
https://www.elastic.co/guide/en/elasticsearch/reference/8.11/query-dsl-terms-query.html
https://stackoverflow.com/questions/28768277/elasticsearch-difference-between-must-and-should-bool-query